addons/mail/static/src/core/common/quick_reaction_menu.js

```diff
@@ -34,7 +34,17 @@ export class QuickReactionMenu extends Component {
                 popoverClass: "o-mail-QuickReactionMenu-pickerPopover",
             }
         );
-        this.dropdown = useState(useDropdownState());
+        this.dropdown = useState(
+            useDropdownState({
+                onClose: () => {
+                    if (this.props.message.thread.messageInEdition) {
+                        this.props.message.thread.messageInEdition.composer.autofocus++;
+                    } else {
+                        this.props.message.thread.composer.autofocus++;
+                    }
+                },
+            })
+        );
```

Comment:
Quick reaction could be triggered in mailboxes. Probably best to put viewing thread in `env` and decide on that `env.thread` rather than the origin thread of the message.